### PR TITLE
P#39193-Replace text domain 'onoffice' to 'onoffice-for-wp-websites'

### PR DESCRIPTION
**Pull request by nglelinh**
P#39193 WP-Plugin und Language-Packs

### CODE REVIEW

File: plugin/Gui/AdminPageBase.php [hunk 1]
```diff
@@ -85,10 +85,10 @@ public function render()
 
 	public function generatePageMainTitle($subTitle)
 	{
-		echo '<h1 class="wp-heading-inline">'.esc_html__('onOffice', 'onoffice');
+		echo '<h1 class="wp-heading-inline">'.esc_html__('onoffice-for-wp-websites', 'onoffice-for-wp-websites');
```
Comment (jayay) [inline plugin/Gui/AdminPageBase.php:88]: This doesn't look right to me

File: plugin/Gui/AdminPageEstateUnitList.php [hunk 2]
```diff
@@ -63,19 +63,19 @@ public function renderContent()
 
 	public function generatePageMainTitle($subTitle)
 	{
-		echo '<h1 class="wp-heading-inline">'.esc_html__('onOffice', 'onoffice');
+		echo '<h1 class="wp-heading-inline">'.esc_html__('onoffice-for-wp-websites', 'onoffice-for-wp-websites');
```
Comment (jayay) [inline plugin/Gui/AdminPageEstateUnitList.php:66]: The first parameter needs to stay

**Review by jayay — CHANGES_REQUESTED** [review]
Most of this is good, but it's missing the fixes I made in [830ad22](https://github.com/onOfficeGmbH/oo-wp-plugin/commit/830ad22b2853ba962f58a5f698cfa8f5d74bb32c). The `.pot` file should be regenerated by running `make pot` (requires GNU Make) and the `.po` files should be updated.

File: tests/WP_UnitTest_Localized.php [hunk 3]
```diff
@@ -78,6 +78,13 @@ protected function switchLocale(string $newLocale)
 	{
 		global $l10n;
 
+        if (!is_textdomain_loaded('onoffice-for-wp-websites') && $newLocale !== 'en_US') {
+            load_textdomain('onoffice-for-wp-websites', ONOFFICE_PLUGIN_DIR.'/languages/onoffice-for-wp-websites-'.$newLocale.'.mo');
```
Comment (jayay) [inline tests/WP_UnitTest_Localized.php:82]: this is a tab/space mix

**Comment by jayay** [comment]
Most translations get loaded, except for a couple and I couldn't find out why. @nglelinh  would you mind having a look? Here's a screenshot. _Show 2 matches_ should be translated as well, but isn't.
image

In addition to that, I don't think the code to load the user defined translations is correct. Would you please try out this?

```diff
--- a/plugin.php
+++ b/plugin.php
@@ -108,15 +108,16 @@ add_action('admin_init', [$pAdminViewController, 'add_ajax_actions']);
 add_action('admin_init', [CaptchaDataChecker::class, 'addHook']);

 add_action('plugins_loaded', function() {
-       load_plugin_textdomain('onoffice', false, basename(ONOFFICE_PLUGIN_DIR).'/languages');
-       load_plugin_textdomain('onoffice-for-wp-websites', false, basename(ONOFFICE_PLUGIN_DIR) . '/languages');
+       load_plugin_textdomain('onoffice-for-wp-websites', false, basename(ONOFFICE_PLUGIN_DIR) . '/languages/onoffice-for-wp-websites');
+       // Check 'onoffice-personalized' Folder exists
+       $onofficePersonalizedFolder = plugin_dir_path(__DIR__) . 'onoffice-personalized/languages';
+       if (is_dir($onofficePersonalizedFolder)) {
+               load_plugin_textdomain('onoffice', false, $onofficePersonalizedFolder);
+       } else {
+               load_plugin_textdomain('onoffice', false, basename(ONOFFICE_PLUGIN_DIR).'/languages');
+       }
 });

-// Check 'onoffice-personalized' Folder exists
-$onofficePersonalizedFolder = plugin_dir_path(__DIR__) . 'onoffice-personalized';
-if (is_dir($onofficePersonalizedFolder)) {
-       load_plugin_textdomain('onoffice', false, basename(ONOFFICE_PLUGIN_DIR) . '/languages');
-}

 // "Settings" link in plugins list
 add_filter('plugin_action_links_'.plugin_basename(__FILE__), [$pAdminViewController, 'pluginSettingsLink']);
```

**Comment by jayay** [comment]
In regards to
> _Show 2 matches_ should be translated as well, but isn't.

I've updated that exact translation on the WP translations platform as it might be possible that's the one and only source for that text domain. I'll have a closer look at that tomorrow.

**Comment by nglelinh** [comment]
> Most translations get loaded, except for a couple and I couldn't find out why. @nglelinh would you mind having a look? Here's a screenshot. _Show 2 matches_ should be translated as well, but isn't.
> image
> 
> In addition to that, I don't think the code to load the user defined translations is correct. Would you please try out this?
> 
> ```diff
> --- a/plugin.php
> +++ b/plugin.php
> @@ -108,15 +108,16 @@ add_action('admin_init', [$pAdminViewController, 'add_ajax_actions']);
>  add_action('admin_init', [CaptchaDataChecker::class, 'addHook']);
> 
>  add_action('plugins_loaded', function() {
> -       load_plugin_textdomain('onoffice', false, basename(ONOFFICE_PLUGIN_DIR).'/languages');
> -       load_plugin_textdomain('onoffice-for-wp-websites', false, basename(ONOFFICE_PLUGIN_DIR) . '/languages');
> +       load_plugin_textdomain('onoffice-for-wp-websites', false, basename(ONOFFICE_PLUGIN_DIR) . '/languages/onoffice-for-wp-websites');
> +       // Check 'onoffice-personalized' Folder exists
> +       $onofficePersonalizedFolder = plugin_dir_path(__DIR__) . 'onoffice-personalized/languages';
> +       if (is_dir($onofficePersonalizedFolder)) {
> +               load_plugin_textdomain('onoffice', false, $onofficePersonalizedFolder);
> +       } else {
> +               load_plugin_textdomain('onoffice', false, basename(ONOFFICE_PLUGIN_DIR).'/languages');
> +       }
>  });
> 
> -// Check 'onoffice-personalized' Folder exists
> -$onofficePersonalizedFolder = plugin_dir_path(__DIR__) . 'onoffice-personalized';
> -if (is_dir($onofficePersonalizedFolder)) {
> -       load_plugin_textdomain('onoffice', false, basename(ONOFFICE_PLUGIN_DIR) . '/languages');
> -}
> 
>  // "Settings" link in plugins list
>  add_filter('plugin_action_links_'.plugin_basename(__FILE__), [$pAdminViewController, 'pluginSettingsLink']);
> ```

@jayay it was about caching on the browser, we retested and see it's working
we also made some fixings on the loading text domain from "onoffice-personalized/languages", please help to recheck again

**Comment by jayay** [comment]
Translations for the German language are now being loaded from the WP platform. It's good to know this works before the changes are even released. This basically means, the plugin doesn't need to ship with the `languages/onoffice-for-wp-websites-de_DE.{po,mo}` files. That's something to care about later, though.

**Review by jayay — APPROVED** [review]

**Pull request merged**